hack/generate-plugin-overview.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2019 The Kubernetes Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"github.com/golang/glog"
+	"github.com/spf13/pflag"
+	"io"
+	"os"
+	"regexp"
+	"sigs.k8s.io/krew/pkg/index"
+	"sigs.k8s.io/krew/pkg/index/indexscanner"
+	"strings"
+)
+
+const (
+	separator  = " | "
+	pageHeader = `# Plugin overview
+
+Below is a table of plugins available in the central krew-index.
+
+`
+)
+
+var (
+	githubRepoPattern = regexp.MustCompile(`.*github.com/([^/]+/[^/#]+)`)
+)
+
+func main() {
+	pluginsDir := pflag.String("plugins-dir", "", "The directory containing the plugin manifests")
+	pflag.Parse()
+
+	if *pluginsDir == "" {
+		pflag.Usage()
+		return
+	}
+
+	plugins, err := indexscanner.LoadPluginListFromFS(*pluginsDir)
+	if err != nil {
+		glog.Fatal(err)
+	}
+
+	out := os.Stdout
+
+	_, _ = fmt.Fprintln(out, pageHeader)
+
+	printTableHeader(out)
+	for _, p := range plugins {
+		printTableRowForPlugin(out, &p)
+	}
+}
+
+func printTableHeader(out io.Writer) {
+	printRow(out, "Name", "Description", "GitHub Stars")
```

Comment:
Can I recommend actually spitting out html tables?
It's often much better to have html tables in .md files as they're more reliable and bring more freedom that probably will be necessary in the future.

Reply:
Sure. However, this also means that we should also translate the markdown used in some of the short-descriptions into html. I think a common choice for that task is blackfriday (used by hugo for example).

Reply:
I think this cmd/foo tool can even have its own go.mod. :)